pkg/scale/bitvec.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale
+
+const (
+	// maxLen equivalent of `ARCH32BIT_BITSLICE_MAX_BITS` in parity-scale-codec
+	maxLen = 268435455
+	// byteSize is the number of bits in a byte
+	byteSize = 8
+)
+
+// BitVec represents rust's `bitvec::BitVec` in SCALE
+// It is encoded as a compact u32 representing the number of bits in the vector
+// followed by the actual bits, rounded up to the nearest byte
+type BitVec interface {
+	// Bits returns the bits in the BitVec
+	Bits() []uint8
+	// Bytes returns the byte representation of the Bits
+	Bytes() []byte
+	// Size returns the number of bits in the BitVec
+	Size() uint
+}
+
+// bitVec implements BitVec
+type bitVec struct {
```

Comment:
I notice that in the `parity-scale-codec` in the `src/bit_vec.rs` ([ref to the code line](https://github.com/paritytech/parity-scale-codec/blob/aaea65ab7b7d0cded3e37f42a819ccedb5db645f/src/bit_vec.rs#LL24C1-L24C1)) all implementations of `Encode` and `Decode` contains a generic called `BitOrder` which is related to least significant bit (Lsb) and Most significant bit (Msb), and in looking in the [test in the file](https://github.com/paritytech/parity-scale-codec/blob/aaea65ab7b7d0cded3e37f42a819ccedb5db645f/src/bit_vec.rs#L90) we can see that it tests using both order types. Should we make the bits ordering a required parameter for encoding/decoding in our implementation?

Reply:
Yes, it can be picked up later when needed. I've updated the issue for this.

Reply:
I don't know. I haven't looked into it since the task is to implement the basic version of `bitvec` which supports only `encode` and `decode` at the moment. 

@timoftime Let me know if this PR needs to be more exhaustive.

Reply:
Ok, I don't think we should extend more this task, but I think we should choose LSB to encode/decode a slice of bits as it looks like Polkadot is using this as the main ordering, see the [query to see the usage of bitvec::order::Lsb0](https://github.com/search?q=repo%3Aparitytech%2Fpolkadot%20bitvec%3A%3Aorder%3A%3ALsb0&type=code), but I didn't found any usage of `Msb0` ordering. So you could highlight that the `bitsToBytes` function uses the `lsb` ordering. 

Just for exemplification here is an example of encoding using both ordering: https://go.dev/play/p/4rCqCZA8lma